### ISSUE TITLE
Error: The Odoo IDE extensions doesn't support this environment yet.

### ISSUE THREAD

**Issue by bobslee** [odoo-ide/vscode-odoo#73]
After installing the extension I get the following error/warning (also see screenshot).
The Odoo IDE extensions doesn't support this environment yet.

I disabled Pylance.
Any suggestions?

I tried to investigate, but the source code is not here on GitHub ?

![image](https://github.com/odoo-ide/vscode-odoo/assets/328803/2f99fde3-0c43-4224-9e66-ea30c367f76c)

**Comment by trinhanhngoc** [comment]
Hello @bobslee , Can you provide information about your VSCode (in About dialog)?

**Comment by bobslee** [comment]
@trinhanhngoc 

Version: 1.87.2
Commit: 863d2581ecda6849923a2118d93a088b0745d9d6
Date: 2024-03-08T15:14:59.643Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Linux x64 6.1.82

**Comment by trinhanhngoc** [comment]
@bobslee , Does the warning appear when you open a project in a remote environment (SSH, etc.)?

**Comment by bobslee** [comment]
@trinhanhngoc No the project (everything) is local in the path provided by Direnv. So no Docker or remote.

**Comment by trinhanhngoc** [comment]
@bobslee , Which linux distro are you using?

**Comment by bobslee** [comment]
@trinhanhngoc NixOs. 
Actually using "nixpkgs" (package manager) can also reproduce this.

**Comment by trinhanhngoc** [comment]
@bobslee , I just created a dev version of the extension to include environment information in the warning message. Can you help me check it?

[vscode-odoo-linux-x64-0.22.6-dev.vsix.zip](https://github.com/odoo-ide/vscode-odoo/files/14854213/vscode-odoo-linux-x64-0.22.6-dev.vsix.zip) (need to unzip)

**Comment by bobslee** [comment]
@trinhanhngoc I'm concerned about including a ZIP of a project whose code is not open source/published.
Where's the actual code?

**Comment by trinhanhngoc** [comment]
@bobslee The project is not open source. You can see the discussion here: https://github.com/odoo-ide/vscode-odoo/issues/28. As an alternative, you can use the official Odoo extension, which is open source: https://marketplace.visualstudio.com/items?itemName=Odoo.odoo

**Comment by bobslee** [comment]
@trinhanhngoc After only installing the vsix and restarting VS Code the following errors (output) occur. 
The following message is gone: `The Odoo IDE extensions doesn't support this environment yet.`

```
2024-04-03 20:21:39.939 [error] Activating extension trinhanhngoc.vscode-odoo failed due to an error:
2024-04-03 20:21:39.939 [error] Error: Pending response rejected since connection got disposed
	at Object.dispose (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:61371)
	at Object.dispose (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:115866)
	at b.handleConnectionClosed (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:116272)
	at b.handleConnectionClosed (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:273602)
	at /home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:116015
	at o.invoke (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:63098)
	at r.fire (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:63863)
	at ne (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:49661)
	at o.invoke (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:63098)
	at r.fire (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:63863)
	at d.fireClose (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:71731)
	at ChildProcess.<anonymous> (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:84025)
	at ChildProcess.emit (node:events:526:35)
	at maybeClose (node:internal/child_process:1091:16)
	at Socket.<anonymous> (node:internal/child_process:449:11)
	at Socket.emit (node:events:514:28)
	at Pipe.<anonymous> (node:net:323:12)
2024-04-03 20:21:39.939 [info] Eager extensions activated
2024-04-03 20:21:39.977 [error] Error: Pending response rejected since connection got disposed
	at Object.dispose (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:61371)
	at Object.dispose (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:115866)
	at b.handleConnectionClosed (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:116272)
	at b.handleConnectionClosed (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:273602)
	at /home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:116015
	at o.invoke (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:63098)
	at r.fire (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:63863)
	at ne (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:49661)
	at o.invoke (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:63098)
	at r.fire (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:63863)
	at d.fireClose (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:71731)
	at ChildProcess.<anonymous> (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:84025)
	at ChildProcess.emit (node:events:526:35)
	at maybeClose (node:internal/child_process:1091:16)
	at Socket.<anonymous> (node:internal/child_process:449:11)
	at Socket.emit (node:events:514:28)
	at Pipe.<anonymous> (node:net:323:12)
2024-04-03 20:21:39.978 [error] Error: Client is not running and can't be stopped. It's current state is: starting
	at b.shutdown (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:111187)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110768)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:273201)
	at b.doInitialize (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110591)
	at async b.start (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:107157)
	at async /home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:35701
	at async t.activate (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:33646)
	at async u.n (/nix/store/2yf2slmb9b87yx98pmvh30xgdzqmz42w-vscode-1.87.2/lib/vscode/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6255)
	at async u.m (/nix/store/2yf2slmb9b87yx98pmvh30xgdzqmz42w-vscode-1.87.2/lib/vscode/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:6218)
	at async u.l (/nix/store/2yf2slmb9b87yx98pmvh30xgdzqmz42w-vscode-1.87.2/lib/vscode/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:140:5675)
2024-04-03 20:21:39.987 [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: false, activationEvent: 'onStartupFinished'
2024-04-03 20:21:39.989 [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
2024-04-03 20:21:40.942 [error] Error: Client is not running and can't be stopped. It's current state is: startFailed
	at b.shutdown (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:111187)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110768)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:273201)
	at b.doInitialize (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110591)
	at async b.start (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:107157)
2024-04-03 20:21:41.698 [error] Error: Client is not running and can't be stopped. It's current state is: startFailed
	at b.shutdown (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:111187)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110768)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:273201)
	at b.doInitialize (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110591)
	at async b.start (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:107157)
2024-04-03 20:21:42.382 [error] Error: Client is not running and can't be stopped. It's current state is: startFailed
	at b.shutdown (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:111187)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110768)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:273201)
	at b.doInitialize (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110591)
	at async b.start (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:107157)
2024-04-03 20:21:43.072 [error] Error: Client is not running and can't be stopped. It's current state is: startFailed
	at b.shutdown (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:111187)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110768)
	at b.stop (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:273201)
	at b.doInitialize (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:110591)
	at async b.start (/home/bob/.vscode/extensions/trinhanhngoc.vscode-odoo-0.22.6-dev/dist/extension.js:1:107157)
```

**Comment by trinhanhngoc** [comment]
@bobslee I am installing NixOS to debug this issue. Do you install vscode using this command: `nix-shell -p vscode`?

**Comment by bobslee** [comment]
@trinhanhngoc See https://search.nixos.org/packages?channel=unstable&show=vscode&from=0&size=50&sort=relevance&type=packages&query=vscode+

Install with:

`nix-env -iA nixos.vscode`

I installed VSCode from the nixpkgs-unstable channel. 
So I always get the latest/rolling release.

**Comment by trinhanhngoc** [comment]
@bobslee , I am getting error messages related to `libstdc++.so.6` in the Odoo IDE log output:

![IMG_1687](https://github.com/odoo-ide/vscode-odoo/assets/11208291/9d2f151c-c263-4be3-8c1c-eb7433bf3849)

What about your Odoo IDE log output?

**Comment by trinhanhngoc** [comment]
@bobslee , Instead of installing `libstdcxx5`, I installed `stdenv.cc.cc.lib` package and with the helpful information in https://discourse.nixos.org/t/what-package-provides-libstdc-so-6/18707 and https://discourse.nixos.org/t/what-is-the-nix-way-to-specify-ld-library-path/6407, I can run VSCode with Odoo IDE extension without any issues.

**Comment by bobslee** [comment]
@trinhanhngoc Thanks for the solution.

### My experience and setup

#### 1. vscode-odoo extension with `stdenv.cc.cc.lib`

After installing and linking `stdenv.cc.cc.lib` into `LD_LIBRARY_PATH` ...
The extension indeed can be installed with the provided VSIX version 0.22.6.
However it doesn't work with version 0.22.5 (same error).

#### 2. Running VC Code and vs-code extension in a Nix shell (with `stdenv.cc.cc.lib`)

Your shared Discourse posts triggered me to run VS Code from a **(Nix) shell**, which has the advantages of per shell configuration.

I maintain an Odoo Nix shell (template) GitHub project which I use to setup and run my Odoo development servers.
https://github.com/novacode-nl/nix-odoo

This avoids to use (containerization) tools like Docker and Gitpod - to run different Odoo versions, packages (Python, PostgreSQL etc). Everything is locally and handled by Nix shell (and also direnv). 

#### vs-code extension and Nix shell fix (stdenv.cc.cc.lib)

Check out the (recent) commit which ensures `stdenv.cc.cc.lib` gets installed and sourced in a Nix shell.
https://github.com/novacode-nl/nix-odoo/commit/af03cb735a584a717792f5ceef27b1003ddd5c98

### Suggestion

So what's the difference between the versions 0.22.5 and 0.22.6 ?

Like me, other people will be wary of closed source software.
Why not open source this and get even more help from a community?
Sorry, I hesitate to use this extension because it's closed source.

**Comment by trinhanhngoc** [comment]
@bobslee I am not sure why you have that problem with version 0.22.5. I installed version 0.22.5 from Marketplace inside VSCode and did not get the warning message `The Odoo IDE extension doesn't support this environment yet` as you did. The only error I got was the one related to `libstdc++.so.6`.

Regarding open source, I understand your concern. I hope I can manage to make it open source at some point in the future.
